docs/benchmarking.md

```diff
@@ -133,10 +254,18 @@ As you can see, the "Total group operation data" block has much higher numbers t
 
 ### Lazy Evaluation
 
-`DebugGroup` does use lazy evaluation, meaning that `compute()` calls are necessary before retrieving tracked operation data, else the operation might have not been executed yet.
-However, `compute()` has been changed to behave like `computeSync()` in that it blocks until the computation is done.
-This is because non-blocking computation can lead to race conditions when printing the result of tracking the group operations, i.e. the computation has not been performed yet when the data is printed.
-So make sure to always call `compute()` on every `DebugGroupElement` before accessing any counter data.
+`DebugGroup` does use lazy evaluation, meaning that you need to ensure all lazy computations have finished before retrieving the tracked results.
+One way to do this is to call `computeSync()` on all operations.
```

Comment:
I mean... don't call it on _all_ operations. Just the values you are interested in, not intermediate ones.